java/src/main/java/com/google/appengine/tools/mapreduce/impl/shardedjob/ShardedJobRunner.java

```diff
@@ -493,6 +509,9 @@ public void run() {
             scheduleTask(jobState, taskState, tx);
             tx.commit();
           }
+        } catch (Throwable e) {
```

Comment:
:warning: watch out, this breaks how the callable with retries work. You are catching but not throwing the error, so It will never retry.